src/framework/mlt_frame.c

```diff
@@ -1133,3 +1161,63 @@ mlt_frame mlt_frame_clone( mlt_frame self, int is_deep )
 
 	return new_frame;
 }
+
+/** Build a planes pointers of image mapping
+ *
+ * For proper and unified planar image processing, planes sizes and planes pointers should
+ * be provides to processing code.
+ *
+ * \public \memberof mlt_frame_s
+ * \param format the image format
+ * \param width width of the image in pixels
+ * \param height height of the image in pixels
+ * \param[in] data pointer to allocated image
+ * \param[out] planes pointers to plane's pointers will be set
+ * \param[out] strides pointers to plane's strides will be set
+ * \return the number of bytes
+ */
+int mlt_image_format_planes( mlt_image_format format, int width, int height, void* data, unsigned char *planes[4], int strides[4])
```

Comment:
I would like some discussion on this function because it feels like it is optimized for ffmpeg libraries.

I would propose we replace this function with three functions: one to return the pointer to a plane, one to return the height and one to return the size of a line. The signatures would be:

```
void* mlt_image_plane_ptr( mlt_image_format format, int width, int height, void* data, size_t plane )
size_t mlt_image_plane_height( mlt_image_format format, int width, int height, void* data, size_t plane )
size_t mlt_image_plane_linesize( mlt_image_format format, int width, int height, void* data, size_t plane )
```

I feel that the discrete functions would be more generally useful throughout MLT. If you want an ffmpeg optimized function, you could create on in the avformat module that provides the behavior of this function using the three functions I propose.

I don't feel strongly about this, but I thought I would throw it out for discussion.